### ISSUE TITLE
Canceled operation causes log message with HTTP 500 and stack dump

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#4941]
## tl;dr

An operation (in this case, a `zed delete` that's canceled mid-way through) generates a log message that users may interpret as more severe than it actually is.

## Details

Repro is with Zed commit 59b72d8.

This issue became apparent when helping the community zync user with what led to opening #4934. We'd just advised them of the workaround described there that snapshot files could safely be deleted by hand. However, given they were working in a production environment, they took an understandably a conservative approach of closely watching the `zed serve` log to confirm that things looked healthy, and that's when they spotted something unrelated that made them pause and ask us for guidance. In their own words:

> I started my file cleanup but I found errors on Zed Serve v1.11 . this error is repeated frequently and it is before I started my clean up.
>```
>2023-12-14T14:33:30.786393327-05:00 {"level":"warn","ts":1702582410.7858477,"logger":"core","msg":"Error","request_id":"2ZXxk5BJMRFXl3e4ZX5XJ8dtaXa","status":500,"error":"context canceled","stacktrace":"[github.com/brimdata/zed/service.(*ResponseWriter).Error](http://github.com/brimdata/zed/service.(*ResponseWriter).Error)\n\t/go/pkg/mod/github.com/brimdata/zed@v1.11.1/service/request.go:261\[ngithub.com/brimdata/zed/service.handleDelete](http://ngithub.com/brimdata/zed/service.handleDelete)\n\t/go/pkg/mod/github.com/brimdata/zed@v1.11.1/service/handlers.go:555\[ngithub.com/brimdata/zed/service.(*Core).authhandle.(*Core).handler.func2](http://ngithub.com/brimdata/zed/service.(*Core).authhandle.(*Core).handler.func2)\n\t/go/pkg/mod/github.com/brimdata/zed@v1.11.1/service/core.go:194\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2136\[ngithub.com/rs/cors.(*Cors).Handler-fm.(*Cors).Handler.func1](http://ngithub.com/rs/cors.(*Cors).Handler-fm.(*Cors).Handler.func1)\n\t/go/pkg/mod/github.com/rs/cors@v1.8.0/cors.go:219\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2136\[ngithub.com/brimdata/zed/service.NewCore.panicCatchMiddleware.func6.1](http://ngithub.com/brimdata/zed/service.NewCore.panicCatchMiddleware.func6.1)\n\t/go/pkg/mod/github.com/brimdata/zed@v1.11.1/service/middleware.go:92\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2136\[ngithub.com/brimdata/zed/service.NewCore.accessLogMiddleware.func5.1](http://ngithub.com/brimdata/zed/service.NewCore.accessLogMiddleware.func5.1)\n\t/go/pkg/mod/github.com/brimdata/zed@v1.11.1/service/middleware.go:72\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2136\[ngithub.com/brimdata/zed/service.NewCore.requestIDMiddleware.func4.1](http://ngithub.com/brimdata/zed/service.NewCore.requestIDMiddleware.func4.1)\n\t/go/pkg/mod/github.com/brimdata/zed@v1.11.1/service/middleware.go:28\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2136\[ngithub.com/gorilla/mux.(*Router).ServeHTTP](http://ngithub.com/gorilla/mux.(*Router).ServeHTTP)\n\t/go/pkg/mod/github.com/gorilla/mux@v1.7.5-0.20200711200521-98cb6bf42e08/mux.go:210\[ngithub.com/brimdata/zed/service.(*Core).ServeHTTP](http://ngithub.com/brimdata/zed/service.(*Core).ServeHTTP)\n\t/go/pkg/mod/github.com/brimdata/zed@v1.11.1/service/core.go:240\nnet/http.serverHandler.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2938\nnet/http.(*conn).serve\n\t/usr/local/go/src/net/http/server.go:2009"}
>```

As the ones that know the code, we're able to look at this and identify it as benign: It's an indication of a canceled HTTP request that was issued through the Zed lake API. Picking apart the stack trace we see the reference to `handleDelete`, so that indicates it's related to a call to the `/delete` endpoint. However, it's probably not reasonable to ask users to piece this together on their own. Instead, the fact it's an HTTP 500 reported as an `Error` at `warn` level with a stack trace makes it look potentially concerning. It's therefore unsurprising that a conservative/production user might stop at this point and ask for our guidance before proceeding, which slows them down operationally.

I don't know the feasibility of being able to catch this specific case of a benign cancel of an operation in progress, but if possible what comes to mind instead would be:

1. A log message at less severe level (`info`?)
2. Flagged with a `msg` other than `Error`
3. Perhaps without the stack trace

## Repro

To repro the what the user saw, I start a `zed serve` and then use the [record-poster](https://github.com/brimdata/zed-perf-testing/tree/main/record-poster) tool to create 10k commits. I then initiate a `zed delete` and hit Control-C right after it starts.

```
$ ./record-poster.py --exitafter 10000
Posting: {"ts": "2023-12-15T20:00:20.210423Z", "message": "twinner"}
Posting: {"ts": "2023-12-15T20:00:20.215971Z", "message": "irresponsible erythropoietic palaeodendrologist corruptible isochromatic oleostearin overword"}

$ zed -version
Version: v1.11.1-27-g59b72d88

$ zed delete -where "ts != null"
^CPost "http://localhost:9867/pool/2ZaqDnpZYArdlkXe9e5Z2J1oWCd/branch/main/delete": interrupted
```

The final two log messages generated by `zed serve` show the HTTP 500 followed by the `Request completed` message for the same `request_id`.

```
$ tail -2 zed.out 
{"level":"warn","ts":1702670554.204098,"logger":"core","msg":"Error","request_id":"2ZaqUDme4vwLQp4bOvK6EFUcg1i","status":500,"error":"context canceled","stacktrace":"github.com/brimdata/zed/service.(*ResponseWriter).Error\n\t/Users/phil/work/zed/service/request.go:261\ngithub.com/brimdata/zed/service.handleDelete\n\t/Users/phil/work/zed/service/handlers.go:555\ngithub.com/brimdata/zed/service.(*Core).authhandle.(*Core).handler.func2\n\t/Users/phil/work/zed/service/core.go:194\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/opt/go/libexec/src/net/http/server.go:2136\ngithub.com/rs/cors.(*Cors).Handler-fm.(*Cors).Handler.func1\n\t/Users/phil/.go/pkg/mod/github.com/rs/cors@v1.8.0/cors.go:219\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/opt/go/libexec/src/net/http/server.go:2136\ngithub.com/brimdata/zed/service.NewCore.panicCatchMiddleware.func6.1\n\t/Users/phil/work/zed/service/middleware.go:92\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/opt/go/libexec/src/net/http/server.go:2136\ngithub.com/brimdata/zed/service.NewCore.accessLogMiddleware.func5.1\n\t/Users/phil/work/zed/service/middleware.go:72\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/opt/go/libexec/src/net/http/server.go:2136\ngithub.com/brimdata/zed/service.NewCore.requestIDMiddleware.func4.1\n\t/Users/phil/work/zed/service/middleware.go:28\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/opt/go/libexec/src/net/http/server.go:2136\ngithub.com/gorilla/mux.(*Router).ServeHTTP\n\t/Users/phil/.go/pkg/mod/github.com/gorilla/mux@v1.7.5-0.20200711200521-98cb6bf42e08/mux.go:210\ngithub.com/brimdata/zed/service.(*Core).ServeHTTP\n\t/Users/phil/work/zed/service/core.go:240\nnet/http.serverHandler.ServeHTTP\n\t/usr/local/opt/go/libexec/src/net/http/server.go:2938\nnet/http.(*conn).serve\n\t/usr/local/opt/go/libexec/src/net/http/server.go:2009"}
{"level":"info","ts":1702670554.204285,"logger":"http.access","msg":"Request completed","request_id":"2ZaqUDme4vwLQp4bOvK6EFUcg1i","host":"localhost:9867","method":"POST","proto":"HTTP/1.1","remote_addr":"[::1]:50385","request_content_length":-1,"url":"/pool/2ZaqDnpZYArdlkXe9e5Z2J1oWCd/branch/main/delete","elapsed":2.771991085,"response_content_length":65,"status_code":500}
```

**Comment by philrz** [comment]
Verified in Zed commit 4c7de04.

Repeating the repro step above:

```
$ zed -version
Version: v1.11.1-30-g4c7de042

$ zed delete -where "ts != null"
^CPost "http://localhost:9867/pool/2ZmtNDsfgsJyKRqubxhQzxOtk3V/branch/main/delete": interrupted
```

The "context canceled" message generated in the `zed serve` log (the first of two below) is now at `info` level, is no longer described as an "error", and doesn't include a stack trace.

```
{"level":"info","ts":1703039146.638428,"logger":"core","msg":"Request context canceled","request_id":"2ZmtaLeDfB1qwrm8NXmDCDdsLw0"}
{"level":"info","ts":1703039146.6384661,"logger":"http.access","msg":"Request completed","request_id":"2ZmtaLeDfB1qwrm8NXmDCDdsLw0","host":"localhost:9867","method":"POST","proto":"HTTP/1.1","remote_addr":"[::1]:55952","request_content_length":-1,"url":"/pool/2ZmtNDsfgsJyKRqubxhQzxOtk3V/branch/main/delete","elapsed":2.494228793,"response_content_length":0,"status_code":200}
```

Thanks @nwt!
